src/systems/filecoin_nodes/clock/clock_subsystem.go

```diff
@@ -9,43 +9,23 @@ var UTCMaxDrift = time.Second
 // source, such as NTP, or a very precise hardware clock.
 var UTCSyncPeriod = time.Hour
 
-// EpochDuration is a constant that represents the UTC time duration
+// EpochDuration is a constant that represents the time in seconds
 // of a blockchain epoch.
-var EpochDuration = time.Second * 15
-
-// ISOFormat is the ISO timestamp format we use, in Go time package notation.
-var ISOFormat = "2006-01-02T15:04:05.999999999Z"
-
-func (_ *UTCClock_I) NowUTC() Time {
-	return Time(time.Now().Format(ISOFormat))
-}
+var EpochDuration = UnixTime(15)
 
 func (_ *UTCClock_I) NowUTCUnix() UnixTime {
 	return UnixTime(time.Now().Unix())
 }
 
-func (_ *UTCClock_I) NowUTCUnixNano() UnixTimeNano {
-	return UnixTimeNano(time.Now().UnixNano())
-}
-
 // EpochAtTime returns the ChainEpoch corresponding to t.
 // It first subtracts GenesisTime, then divides by EpochDuration
-// and returns the resulting number of epochs.
-func (c *ChainEpochClock_I) EpochAtTime(t Time) ChainEpoch {
-	g1 := c.GenesisTime()
-	g2, err := time.Parse(ISOFormat, string(g1))
-	if err != nil {
-		// an implementation should probably not panic here
-		// this is for simplicity of the spec
-		panic(err)
-	}
-
-	t2, err := time.Parse(ISOFormat, string(t))
-	if err != nil {
-		panic(err)
+// and returns the resulting number of epochs. If t is before
+// GenesisTime zero is returned.
+func (c *ChainEpochClock_I) EpochAtTime(t UnixTime) ChainEpoch {
+	if t <= c.GenesisTime() {
+		return ChainEpoch(0)
```

Comment:
No, I think this is better to return negative numbers, so that the first (zero) epoch is distinguishable from negatives. Calling code can deal with negatives if possible in that context.